src/updateAssetURLForGQL.ts

```diff
@@ -0,0 +1,26 @@
+export function updateAssetURLForGQL(gqlResponse:any) {
+  try {
+    const response = gqlResponse.data;
+    for (let contentType in response) {
+      const entry = response[contentType]; // page_json_rte
+      for (let key in entry) {
+        const field = entry[key];
+        if (field && field.embedded_itemsConnection) { // rte field with embedded items e.g rte_2
+          const edges = field.embedded_itemsConnection.edges;
+          edges.forEach((edge:any) => {
+            const node = edge.node;
+            if (node.url && node.filename) {
+              
+              if (!node?.system?.uid) throw new Error('Asset UID not found in the response');
+
+              const correspondingAsset = field.json.children.find((child:any) => child.attrs['asset-uid'] === node.system.uid);
+              correspondingAsset.attrs['asset-link'] = node.url;
+            }
+          });
+        }
+      }
+    }
+  } catch (error) {
+    console.error('Error in updating asset URL for GQL response', error);
+  }
+}
```

Comment:
@nadeem-cs , [Minor] If it's urgent please ignore below points.

We can define type definitions to to represent the structure of the data. This helps with type checking and readability.
And also instead of keeping code in one function we can separate it out like updateContentType, updateField
interface Node {
  url?: string;
  filename?: string;
  system?: {
    uid?: string;
  };
}

interface Edge {
  node: Node;
}

interface GQLResponse {
  data: {
    [key: string]: {
      [key: string]: Field;
    };
  };
}

Reply:
Thanks for the feedback @aman19K 
Will be ignoring this now as this is an urgent release and this is only a workaround we're providing to a specific customer